bench/StressBench.java

```diff
@@ -0,0 +1,107 @@
+package benchmark;
+
+import com.sun.cldchi.jvm.JVM;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import javax.microedition.io.Connector;
+import javax.microedition.io.SocketConnection;
+import javax.microedition.midlet.MIDlet;
+import javax.microedition.lcdui.Canvas;
+import javax.microedition.lcdui.Display;
+import javax.microedition.lcdui.Graphics;
+
+// This needs to be a midlet in order to have access to the J2ME socket API.
+
+public class StressBench extends MIDlet {
+
+  static class BouncyColors extends Canvas {
+    int x = 0;
+    int y = 0;
+    int dx = 3;
+    int dy = 5;
+
+    public void paint(Graphics g) {
+      int z = 0;
+      while (true) {
+        if ((x + dx) < 0 || (x + dx) > 200) {
+          dx *= -1;
+        }
+        if ((y + dy) < 0 || (y + dy) > 200) {
+          dy *= -1;
+        }
+        x += dx;
+        y += dy;
+        g.setColor(255, 255, 255);
+        g.fillRect(0, 0, 300, 300);
+
+        g.setColor((z + x) & 0xff, (z + y) & 0xff, (x + y) & 0xff);
+        g.fillRect(x, y, 10, 10);
+        try {
+          Thread.sleep(16);
+        } catch (Exception x) {
+          // ...
+        }
+      }
+    }
+  }
+
+  static class Worker implements Runnable {
+
+    void benchmarkLargeRead() throws IOException {
+      SocketConnection client = (SocketConnection)Connector.open("socket://localhost:8000");
+
+      OutputStream os = client.openOutputStream();
+      os.write("GET /bench/benchmark.jar HTTP/1.1\r\nHost: localhost\r\n\r\n".getBytes());
```

Comment:
Adding "Connection: close\r\n" here makes this test work with real webservers that keep sockets open for http 1.1.